.github/workflows/ansible-test.yml

```
@@ -13,15 +28,19 @@ jobs:
           path: ansible_collections/community/sops
 
       - name: Set up Python 3.6
-        uses: actions/setup-python@v1
+        uses: actions/setup-python@v2
         with:
-          python-version: 3.6
+          python-version: ${{ matrix.python }}
 
-      - name: Install ansible-base (devel)
-        run: pip install https://github.com/ansible/ansible/archive/devel.tar.gz --disable-pip-version-check
+      - name: Install ansible-base (${{ matrix.ansible }})
+        run: pip install https://github.com/ansible/ansible/archive/${{ matrix.ansible }}.tar.gz --disable-pip-version-check
 
+      # run ansible-test sanity inside of Docker.
+      # The docker container has all the pinned dependencies that are required.
+      # Explicity specify the version of Python we want to test
       - name: Run sanity tests
-        run: ansible-test sanity --docker -v --color --python 3.6
+        run: ansible-test sanity --docker -v --color --python ${{ matrix.python }}
```

The `--python` option should not be used for sanity tests when `--docker` is there. (And you can fix the Python version to 3.8 in the Python setup above.)
```suggestion
        run: ansible-test sanity --docker -v --color
```